observability/tracing.go

```diff
@@ -0,0 +1,119 @@
+// The observability package provides utilities for the Kiali server
+// to instrument itself with observability tools such as tracing to provide
+// better insights into server performance.
+package observability
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/kiali/kiali/config"
+)
+
+const (
+	// TracingService is the name of the kiali tracer service.
+	TracingService = "kiali"
+)
+
+// EndFunc ends a span if one is started. Otherwise does nothing.
+type EndFunc func()
+
+// TracerName is the name of the global kiali Trace.
+func TracerName() string {
+	return TracingService + "." + config.Get().Deployment.Namespace
+}
+
+// InitTracer initalizes a TracerProvider that exports to jaeger.
+// This will panic if there's an error in setup.
+func InitTracer(jaegerURL string) *sdktrace.TracerProvider {
+	exporter, err := jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(jaegerURL)))
+	if err != nil {
+		panic(err)
+	}
+
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.ParentBased(sdktrace.TraceIDRatioBased(0.5))), // Sample half of traces.
```

Comment:
This parameter "0.5" is pretty sensible to performance impact. I'd suggest making it available in the configuration.